pageserver/src/storage_sync.rs

```diff
@@ -920,6 +966,7 @@ impl RemoteTimelineClient {
             // Launch any queued tasks that were unblocked by this one.
             self.launch_queued_tasks(upload_queue);
         }
+        self.update_upload_queue_unfinished_metric(-1, &task.op);
```

Comment:
This operation might be dropped if the `select!` in the caller cancels us in favor of the shutdown watcher future.
Wonder why the test didn't catch it.

Similar concern for the accounting in `num_*` above, but less relevant, since that state will be dropped shortly after anyways. But the metric state remains and must be correct. It's a gauge.